csrc/host_ir/lower_to_llvm.cpp

```diff
@@ -0,0 +1,912 @@
+#include <ops/all_ops.h>
+#include <val_graph_visitor.h>
+
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Support/raw_ostream.h"
+#include "llvm/IR/Verifier.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/TargetSelect.h"
+#include "llvm/IR/LLVMContext.h"
+#include <llvm/ExecutionEngine/Orc/ThreadSafeModule.h>
+#include <llvm/ExecutionEngine/Orc/LLJIT.h>
+#include <llvm/ExecutionEngine/Orc/CompileUtils.h>
+#include <llvm/ExecutionEngine/Orc/IRCompileLayer.h>
+#include <llvm/ExecutionEngine/Orc/RTDyldObjectLinkingLayer.h>
+#include <llvm/ExecutionEngine/JITLink/JITLink.h>
+#include "llvm/Support/Error.h"
+
+#include <unordered_map>
+#include <queue>
+#include <chrono>
+
+#include <host_ir/lower_to_llvm.h>
+
+namespace nvfuser {
+/*
+
+Helper Data Structures & Functions
+
+*/
+
+using ShapeInferFunc = void (*)(const int64_t* input_tensor_shape, int64_t input_tensor_shape_buffer_size, 
+int64_t* output_tensor_shape, int64_t output_tensor_shape_buffer_size);
+
+using StrideInferFunc = void (*)(const int64_t* output_tensor_shape, int64_t output_tensor_shape_buffer_size, int64_t* output_tensor_stride, 
+int64_t output_tensor_stride_buffer_size, int64_t* sharded_output_tensor_shape, int64_t sharded_output_tensor_shape_buffer_size);
+
+// Dependency graph entry for the stride inference
+struct StrideInfo {
+public:
+    llvm::Value* llvm_extent = nullptr;   // LLVM Value for the extent of this IterDomain
+    llvm::Value* llvm_stride = nullptr;   // LLVM Value for the calculated stride of this IterDomain
+};
+
+// Helper function to exit on error on LLVM JIT initialization
+template <typename T>
+T ExitOnErr(llvm::Expected<T> &&E) {
+    if (!E) {
+        NVF_ERROR(false, "LLVM JIT Initialization Error: " + llvm::toString(E.takeError()));
+        llvm::errs() << llvm::toString(E.takeError()) << "\n";
+        exit(1);
+    }
+    return std::move(*E);
+}
+
+inline void ExitOnErr(llvm::Error &&Err) {
+  if (Err) {
+    NVF_ERROR(false, "LLVM JIT Initialization Error: " + llvm::toString(std::move(Err)));
+    llvm::errs() << llvm::toString(std::move(Err)) << "\n";
+    exit(1);
+  }
+}
+
+// Helper function to cast iter domains to vals
+std::vector<Val*> domain2vals(const std::vector<IterDomain*>& domain){
+  std::vector<Val*> vals;
+  for(auto* id : domain){
+    vals.push_back(dynamic_cast<Val*>(id));
+  }
+  return vals;
+}
+
+// Helper function to map the current domain to the input domain strictly
+int isSameToInputDomain(std::unordered_map<int, Val*>& boundary_vals, Val* current_domain){
+  for(auto boundary_val : boundary_vals){
+    if(boundary_val.second == current_domain){
+      return boundary_val.first;
+    }
+  }
+  return -1;
+}
+
+// Helper function to check if the current iter domain is alias to the input iter domain
+int mapToInputDomain(std::unordered_map<int, Val*>& boundary_vals, Val* current_domain, const ValGraph& exact_graph){
```

Comment:
This function should be renamed, you are mapping things to your boundary_vals, it's not necessarily an input.

Also, we are returning the mapping index, it doesn't need to be an alias.